lib/Mojolicious/Guides/Tutorial.pod

```diff
@@ -0,0 +1,938 @@
+
+  use Mojolicious::Lite -signatures;
+
+  # Route leading to an action that renders some text
+  get '/foo' => sub ($c) {
+    $c->render(text => 'Hello World!');
+  };
+
+  app->start;
+
+Response content is often generated by actions with L<Mojolicious::Controller/"render">, but more about that later.
+
+=head2 GET/POST parameters
+
+All C<GET> and C<POST> parameters sent with the request are accessible via L<Mojolicious::Controller/"param">.
+
+  use Mojolicious::Lite -signatures;
+
+  # /foo?user=sri
+  get '/foo' => sub ($c) {
+    my $user = $c->param('user');
+    $c->render(text => "Hello $user.");
+  };
+
+  app->start;
+
+=head2 Stash and templates
+
+The L<Mojolicious::Controller/"stash"> is used to pass data to templates, which can be inlined in the C<DATA> section.
+A few stash values like C<template>, C<text> and C<data> are reserved and will be used by
+L<Mojolicious::Controller/"render"> to decide how a response should be generated.
+
+  use Mojolicious::Lite -signatures;
+
+  # Route leading to an action that renders a template
+  get '/foo' => sub ($c) {
+    $c->stash(one => 23);
+    $c->render(template => 'magic', two => 24);
+  };
+
+  app->start;
+  __DATA__
+
+  @@ magic.html.ep
+  The magic numbers are <%= $one %> and <%= $two %>.
+
+For more information about templates see also L<Mojolicious::Guides::Rendering/"Embedded Perl">.
+
+=head2 HTTP
+
+L<Mojolicious::Controller/"req"> and L<Mojolicious::Controller/"res"> give you full access to all HTTP features and
+information.
+
+  use Mojolicious::Lite -signatures;
+
+  # Access request information
+  get '/agent' => sub ($c) {
+    my $host = $c->req->url->to_abs->host;
+    my $ua   = $c->req->headers->user_agent;
+    $c->render(text => "Request by $ua reached $host.");
+  };
+
+  # Echo the request body and send custom header with response
+  post '/echo' => sub ($c) {
+    $c->res->headers->header('X-Bender' => 'Bite my shiny metal ass!');
+    $c->render(data => $c->req->body);
+  };
+
+  app->start;
+
+You can test the more advanced examples right from the command line with L<Mojolicious::Command::get>.
+
+  $ ./myapp.pl get -v -M POST -c 'test' /echo
+
+=head2 JSON
+
+JSON is the most commonly used data-interchange format for web services. L<Mojolicious> loves JSON and comes with the
+possibly fastest pure-Perl implementation L<Mojo::JSON> built right in, which is accessible through
+L<Mojo::Message/"json"> as well as the reserved stash value C<json>.
+
+  use Mojolicious::Lite -signatures;
+
+  # Modify the received JSON document and return it
+  put '/reverse' => sub ($c) {
+    my $hash = $c->req->json;
+    $hash->{message} = reverse $hash->{message};
+    $c->render(json => $hash);
+  };
+
+  app->start;
+
+You can send JSON documents from the command line with L<Mojolicious::Command::get>.
+
+  $ ./myapp.pl get -M PUT -c '{"message":"Hello Mojo!"}' /reverse
+
+=head2 Built-in C<exception> and C<not_found> pages
```

Comment:
Would be great to link here how or even just that these built-in pages can be modified.
A great example for the cookbook would be using a hook to detect when these pages are rendered so that special log entries are generated or perhaps an alert such as a webhook is created.  Likely this isn't so important for a 404, but for a 500 most admins would want to know ASAP.